ouroboros-consensus/src/Ouroboros/Consensus/HardFork/History/Qry.hs

```diff
@@ -558,6 +562,36 @@ epochToSizeExpr :: EpochNo -> Expr f EpochSize
 epochToSizeExpr absEpoch =
     EEpochSize (ELit absEpoch)
 
+{-------------------------------------------------------------------------------
+  Conversion to the related EpochInfo interface
+-------------------------------------------------------------------------------}
+
+-- | The 'EpochInfo' interface provides similar functionality to the history
+-- interpreter.
+--
+-- The history interpreter can be viewed through the 'EpochInfo' interface.
+--
+toEpochInfo :: Interpreter xs -> EpochInfo (Either PastHorizonException)
```

Comment:
At a glance, this seems to have a lot of overlap with `\(Interpreter summary) -> snapshotEpochInfo summary`; I would anticipate we could at least define one via the other.

`snapshotEpochInfo` is from `Ouroboros/Consensus/HardFork/History/EpochInfo.hs`

Reply:
The way it is shows it's definable via the public API of the interpreter. If we don't care about that then yes the direct way is a lot less code.